### PR TITLE
Saved php types with metadata when fetching

### PR DESCRIPTION
**Pull request by yitam**

### CODE REVIEW

**Comment by david-puglielli** [comment]
This looks good to me, but do we have tests that could ferret out issues with caching php types?

**Comment by yitam** [comment]
@david-puglielli many existing fetching tests go through this path, but let me see if I can add more tests.

File: source/pdo_sqlsrv/pdo_stmt.cpp [hunk 1]
```diff
@@ -782,14 +782,11 @@ int pdo_sqlsrv_stmt_get_col_data( _Inout_ pdo_stmt_t *stmt, _In_ int colno,
 
         // set the encoding if the user specified one via bindColumn, otherwise use the statement's encoding
         // save the php type for next use
-        if (driver_stmt->current_meta_data[colno]->sqlsrv_php_type.typeinfo.type != SQLSRV_PHPTYPE_INVALID) {
-            sqlsrv_php_type = driver_stmt->current_meta_data[colno]->sqlsrv_php_type;
-        }
```
Comment (david-puglielli) [inline source/pdo_sqlsrv/pdo_stmt.cpp]: Did this cause problems with the new tests?

Comment (yitam) [reply source/pdo_sqlsrv/pdo_stmt.cpp]: Yes, as shown in the pdo test, the user may choose different encoding or type for every single fetch

**Pull request merged**